src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/PSUserAgent.cs

```diff
@@ -118,19 +122,44 @@ internal static string App
             }
         }
 
-        internal static string Platform
+        internal static string PlatformName
         {
             get
             {
-                return ("Windows NT");
+                if (Platform.IsWindows)
+                {
+                    // only generate the windows user agent once
+                    if(PSUserAgent._windowsUserAgent == null){
+                        // find the version in the windows operating system description
+                        string versionText = PSUserAgent.OS.Substring(PSUserAgent.OS.TrimEnd().LastIndexOf(" ") +1);
```

Comment:
Do we really need `TrimEnd()`?
If so it is better mobe it in OS property.

Please remove 'PSUserAgent' prefix.